src/Compilers/Core/Portable/StrongName/StrongNameKeys.cs

```diff
@@ -113,7 +113,7 @@ internal static StrongNameKeys Create(string keyFilePath, CommonMessageProvider
         //In IDE typing scenarios we often need to infer public key from the same
         //key file blob repeatedly and it is relatively expensive.
         //So we will store last seen blob and corresponding key here.
-        private static Tuple<ImmutableArray<byte>, ImmutableArray<byte>> s_lastSeenKeyPair;
+        private static Tuple<ImmutableArray<byte>, ImmutableArray<byte>, RSAParameters?> s_lastSeenKeyPair;
```

Comment:
>Tuple<ImmutableArray<byte>, ImmutableArray<byte>, RSAParameters?> [](start = 23, length = 65)

Can we use the tuple syntax with named members??

Reply:
We need to use `System.Tuple` instead of `System.ValueTuple` because of the interlocked exchange in `CreateHelper` down below.